server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -253,7 +252,7 @@ protected static Collection<String> resolveExpressions(Context context, String..
             } else {
                 return ExplicitResourceNameFilter.filterUnavailable(
                     context,
-                    DateMathExpressionResolver.resolve(context, List.of(expressions))
+                    DateMathExpressionResolver.resolve(context, Arrays.asList(expressions))
```

Comment:
This is a pretty big deal in some cases, `List.of` will copy the expressions list which is quite heavy once we get to very long lists.

Reply:
You could `Collections.unmodifiableList(...)` the list from `Arrays.asList` if you wanted. It's an allocation and adds some overhead to the list operations, but I don't think it's nearly as much work as copying the entire list.

Reply:
I was actually thinking about dropping the `Arrays` wrapping in a follow-up :P that would make this perform yet a little nicer actually :D